cfme/tests/infrastructure/test_child_tenant.py

```diff
@@ -0,0 +1,133 @@
+    field, value = request.param
+    new_child.set_quota(**{'{}_cb'.format(field): True, field: value})
+    yield
+    # will refresh page as navigation to configuration is blocked if alerts are on the page
+    appliance.server.login_admin()
+    appliance.server.browser.refresh()
+    new_child.set_quota(**{'{}_cb'.format(field): False})
+
+
+@pytest.fixture(scope='module')
+def new_tenant(appliance):
+    collection = appliance.collections.tenants
+    tenant = collection.create(name='tenant{}'.format(fauxfactory.gen_alphanumeric()),
+                               description='tenant_des{}'.format(fauxfactory.gen_alphanumeric()),
+                               parent=collection.get_root_tenant())
+    yield tenant
+    tenant.delete()
+
+
+@pytest.fixture(scope='module')
+def new_child(appliance, new_tenant):
+    collection = appliance.collections.tenants
+    child_tenant = collection.create(name='tenant{}'.format(fauxfactory.gen_alphanumeric()),
+                                     description='tenant_des{}'.format(
+                                         fauxfactory.gen_alphanumeric()),
+                                     parent=new_tenant)
+    yield child_tenant
+    child_tenant.delete()
+
+
+@pytest.fixture(scope='module')
+def new_group(appliance, new_child, new_tenant):
+    collection = appliance.collections.groups
+    group = collection.create(description='group_{}'.format(fauxfactory.gen_alphanumeric()),
+                              role='EvmRole-super_administrator',
+                              tenant='My Company/{}/{}'.format(new_tenant.name, new_child.name))
+    yield group
+    group.delete()
+
+
+@pytest.fixture(scope='module')
+def new_user(appliance, new_group, new_credential):
+    collection = appliance.collections.users
+    user = collection.create(
+        name='user_{}'.format(fauxfactory.gen_alphanumeric()),
+        credential=new_credential,
+        email='xyz@redhat.com',
+        groups=new_group,
+        cost_center='Workload',
+        value_assign='Database')
+    yield user
+    user.delete()
+
+
+@pytest.mark.rhv2
+# first arg of parametrize is the list of fixtures or parameters,
+# second arg is a list of lists, with each one a test is to be generated
+# sequence is important here
+# indirect is the list where we define which fixtures are to be passed values indirectly.
+@pytest.mark.parametrize(
+    ['set_child_tenant_quota', 'custom_prov_data', 'extra_msg', 'approve'],
+    [
+        [('cpu', '2'), {'hardware': {'num_sockets': '8'}}, '', False],
+        [('storage', '0.01'), {}, '', False],
+        [('memory', '2'), {'hardware': {'memory': '4096'}}, '', False],
+        [('vm', '1'), {'catalog': {'num_vms': '4'}}, '###', True]
+    ],
+    indirect=['set_child_tenant_quota'],
+    ids=['max_cpu', 'max_storage', 'max_memory', 'max_vms']
+)
+def test_child_tenant_quota_enforce_via_lifecycle_infra(appliance, provider, setup_provider,
```

Comment:
@anurag03 This looks like can be moved to https://github.com/ManageIQ/integration_tests/pull/6944  also fixtures duplication can be omitted
Any reason to be in a separate file?

Reply:
@rlbabyuk : Actually I have to write separate fixture for user, group and tenant for both project and child, so it will be very confusing and again repetitive in  same file, so I wanted to distinguish them, hence I have used different file for it. I can move them in similar PR but in same file I don't want to create the confusion as there are too many fixtures.